idaes/models_extra/power_generation/costing/power_plant_capcost.py

```diff
@@ -0,0 +1,2015 @@
+    - display_flowsheet_cost() to display flowsheet cost
+    - check_sCO2_costing_bounds() to display a warning if costing model have
+      been used outside the range that where designed for
+"""
+__author__ = "Costing Team (A. Noring, B. Paul, D. Caballero, and M. Zamarripa)"
+__version__ = "1.0.0"
+
+from pandas import DataFrame
+from sys import stdout
+import textwrap
+
+from pyomo.environ import (
+    Param,
+    Var,
+    Constraint,
+    Expression,
+    value,
+    Expr_if,
+    units as pyunits,
+)
+from pyomo.core.base.expression import ScalarExpression
+from pyomo.core.base.units_container import InconsistentUnitsError
+from pyomo.util.calc_var_value import calculate_variable_from_constraint
+
+import idaes.core.util.scaling as iscale
+from idaes.core import (
+    FlowsheetCostingBlockData,
+    register_idaes_currency_units,
+)
+from idaes.models_extra.power_generation.costing.costing_dictionaries import (
+    BB_costing_params,
+    sCO2_costing_params,
+)
+from idaes.models_extra.power_generation.costing.generic_ccs_capcost_custom_dict import (
+    generic_ccs_costing_params,
+)
+
+from idaes.core.util.tables import stream_table_dataframe_to_string
+
+import idaes.logger as idaeslog
+
+import pyomo.environ as pyo
+from idaes.core import declare_process_block_class
+
+_log = idaeslog.getLogger(__name__)
+
+# -----------------------------------------------------------------------------
+# Power Plant Costing Library
+# -----------------------------------------------------------------------------
+
+
+@declare_process_block_class("QGESSCosting")
+class QGESSCostingData(FlowsheetCostingBlockData):
+    # Register currency and conversion rates based on CE Index
+    register_idaes_currency_units()
+
+    def build_global_params(self):
+        """
+        This is where we can declare any global parameters we need, such as
+        Lang factors, or coefficients for costing methods that should be
+        shared across the process.
+
+        You can do what you want here, so you could have e.g. sub-Blocks
+        for each costing method to separate the parameters for each method.
+        """
+        # Set the base year for all costs
+        self.base_currency = pyo.units.USD_2018
+        # Set a base period for all operating costs
+        self.base_period = pyo.units.year
+
+    def build_process_costs(
+        self,
+        # arguments related to Fixed OM costs
+        total_plant_cost=None,
+        nameplate_capacity=650,
+        labor_rate=38.50,
+        labor_burden=30,
+        operators_per_shift=6,
+        tech=1,
+        # arguments related to total owners costs
+        land_cost=None,
+        net_power=None,
+        resources=None,
+        rates=None,
+        prices=None,
+        fixed_OM=True,
+        variable_OM=False,
+        fuel=None,
+        chemicals=None,
+        waste=None,
+        tonne_CO2_capture=None,
+        CE_index_year="2018",
+    ):
+        """
```

Comment:
You should probably update this to be more specific to the current case. This looks like it is just taken from the base class/example.

For one, you should document all the arguments to this function in the doctstring (which can then be used to automatically generate docs for the method as well).